cockpit/leapp.html

```diff
@@ -226,12 +226,34 @@ <h3 class="mapped_ports">Mapped ports - Source &rarr; Target</h3>
                     $("#source-machines ul").append(machine_html.replace(/\[type\]/g, "source"))
                 }
             }
+            $("#target-machines ul").append(`
+                <li>
+                    <input id="machine-radio-custom-target" class="radio-target-machine ui-lockable"
+                    type="radio" name="target-machine" value="" data-ip=""/>
+                    <label for="machine-radio-custom-target">Custom Target</label>
+                    <input type="text" class="form-control" id="custom_target" name="custom_target" placeholder="IP or Fully qualified domain name"/>
+                </li>
+            `);
             output.empty();
             if (vm_data.machines.length == 0) {
                 output.append(document.createTextNode("No running local VMs!\n"));
             }
             loader.hide();
 
+            $("input[name=target-machine]").change(function() {
+                if ($("#machine-radio-custom-target").is(":checked")) {
+                    $("#custom_target").show();
+                } else {
+                    $("#custom_target").hide();
+                }
+            });
+            $("#custom_target")
+                .toggle()
+                .focusout(get_target_ports)
+                .change(function() {
+                    $("#machine-radio-custom-target").val($("#custom_target").val());
+                    $("#machine-radio-custom-target").data('ip', $("#custom_target").val());
+            });
```

Comment:
one more thing, it would be good to add ```$("#machine-radio-custom-target").unbind('click');``` after
```$(".radio-target-machine").on('click..``` to prevent situation like scanning target ports without filled address